routemaster/state_machine/api.py

```diff
@@ -207,13 +201,10 @@ def delete_label(app: App, label: LabelRef) -> None:
     # Record the label as having been deleted and remove its metadata
     row.metadata = {}
     row.deleted = True
-    app.session.add(row)
 
     # Add a history entry for the deletion
     current_state = get_current_state(app, label, state_machine)
-    app.session.add(History(
-        label_name=label.name,
-        label_state_machine=label.state_machine,
+    row.history.append(History(
```

Comment:
So the relationship is smart enough to figure out the things that can be set based on the foreign key constraint?